### PR TITLE
Update API contract for new Library deployment

### PR DESCRIPTION
**Pull request by nvisal1**
***Purpose***
This PR changes 3 route contracts in the Library Service.
Library Item save and delete operations will no longer return an updated list of Library Items
Library Item read with pagination will now include a `lastPage` property in the response.

***Implementation***
Library Item save now returns a status of 201
Library Item delete now returns a status of 204

***Story***
This PR addresses Clubhouse story https://app.clubhouse.io/clarkcan/story/1646/library-redesign-deployment

### CODE REVIEW

File: src/drivers/http/ExpressAuthRouteDriver.ts [hunk 1]
```diff
@@ -82,15 +85,9 @@ export class ExpressAuthRouteDriver {
...
-    const limit = parseInt(req.query.limit, 10);
     try {
-      const userLibraryItems = await removeItemFromUserLibrary({ username: providedUsername, cuid, requester, page, limit });
-      userLibraryItems.map(
-        (userLibraryItem: any) => {
-          (userLibraryItem.learningObject = (new LearningObject(userLibraryItem.learningObject).toPlainObject()))
-        });
-      res.status(200).json(userLibraryItems);
+      await removeItemFromUserLibrary({ username: providedUsername, cuid, requester });
+      res.sendStatus(204);
```
Comment (pzalep1) [inline src/drivers/http/ExpressAuthRouteDriver.ts]: On the frontend, are we now just going to make a request to get the library again after we get this back?

File: src/drivers/http/ExpressAuthRouteDriver.ts [hunk 2]
```diff
@@ -51,7 +52,13 @@ export class ExpressAuthRouteDriver {
         (userLibraryItem: any) => {
           (userLibraryItem.learningObject = (new LearningObject(userLibraryItem.learningObject).toPlainObject()))
         });
-      res.status(200).json(userLibraryItems);
+      
+      if (page && limit) {
```
Comment (EVogel1999) [inline src/drivers/http/ExpressAuthRouteDriver.ts]: If we only use limit in this if block, why are we also checking for page?

File: src/drivers/http/ExpressAuthRouteDriver.ts (hunk 1 above)
Comment (nvisal1) [reply src/drivers/http/ExpressAuthRouteDriver.ts]: Yes

File: src/drivers/http/ExpressAuthRouteDriver.ts (hunk 2 above)
Comment (nvisal1) [reply src/drivers/http/ExpressAuthRouteDriver.ts]: Both `page` and `limit` must be passed in order for pagination to work correctly. I put the check here to allow for temporary backward compatibility with the old Library. This route should throw a 400 if both are not passed after the new Library deployment.

**Pull request merged**